MPU/include/mpu.h

```diff
@@ -11,158 +11,92 @@
+ * 
+ * @param status 
+ */
+void writeFaultLatch(bool status);
 
-        /**
-         * @brief Handles all Pedals Processes
-         * 
-         */
-        void pedalsProcess();
+/**
+* @brief Shuts off the car in the event of an error
+* @note ****Code stays here until power cycle for safety****
+* 
+*/
+void shutOffCar();
         
-        /**
-         * @brief Handles all GPIO Processes
-         * 
-         */
-        void gpioProcess();
-
-        /**
-         * @brief Sends the message that was loaded into the MC
-         * 
-         */
-        void sendMCMsg();
-
-        /**
-         * @brief Passes loaded temp into private BMS object
-         * 
-         */
-        void setBMSAvgTemp(uint8_t p_avgTemp);
-
-        /**
-         * @brief Passes Loaded SoC into private BMS object
-         * 
-         */
-        void setBMSSoC(uint8_t p_soc);
-
-        /**
-         * @brief Checks if the CAN line is intact
-         * 
-         */
-        bool isCANLineOK();
-
-        /**
-         * @brief Sets the canLineOK to true
-         * 
-         */
-        void CANLineVerified();
-
-        /**
-         * @brief Shuts down the car if the isShutdown boolean is false
-         * 
-         */
-        void checkShutdownStatus();
-
-        /**
-         * @brief sets the current draw limit for the BMS
-         * 
-         */
-        void setCurrentLimit(uint16_t currentLimit);
-
-        /**
-         * @brief Set the Charge Current Limit for the BMS
-         * 
-         * @param currentLimit 
-         */
-        void setChargeCurrentLimit(uint16_t currentLimit);
-
-        /**
-         * @brief Sets the current draw and also puts the BMS into boost mode if the current exceeds the current limit
-         * 
-         */
-        void bmsCurrentProcess(int16_t currentDraw);
+/**
+ * @brief Handles all driverio processes
+ * 
+ */
+void driverioProcess();
+
+/**
+ * @brief Handles all Pedals Processes
+ * 
+ */
+void pedalsProcess();
         
-        /**
-         * @brief Set the motor speed of the MC
-         * 
-         * @param motorSpeed 
-         */
-        void setMotorSpeed(int16_t motorSpeed);
-
-        /**
-         * @brief Enables the BMS to stay in Charging Mode
-         * 
-         */
-        void enableBMSChargingMode();
-
-        /**
-         * @brief Sets the Live BMS Voltage
-         * 
-         */
-        void setBMSVoltage(int16_t voltage);
-
-        /**
-         * @brief Set the Motor Temp
-         * 
-         * @param temp 
-         */
-        void setMotorTemp(int16_t temp);
-
-        /**
-         * @brief Make sure the motor is spinning if we send a motor torque command
-         * 
-         * @return true 
-         * @return false 
-         */
-        bool verifyMotorSpinning();
-};
+/**
+ * @brief Handles all GPIO Processes
+ * 
+ */
+void gpioProcess();
+
+/**
+ * @brief Checks if the CAN line is intact
+ * 
+ */
+bool isCANLineOK();
```

Comment:
Generally speaking can we condense these functions. For example, this CAN line check. Rn we have CANLineVerified, isCANLineOK, and so on. Can we not simplify and just make a "CANStatus" function that returns the status and also takes the status as a parameter, and when a parameter is provided it updates, and when not it just returns the current value?

Reply:
I feel like this CANLine function exemplifies the type of bloated stuff that makes me worried. When I want to debug the car I want to be able to debug chunks, not functions that link to functions that link to functions that each contain a single actual line of code

Reply:
I may also be way off on this and it's just I'm taking an EE stance and you are taking the CE/CS stance, but worth a consideration as we rework all of this